### PR TITLE
Pass agent-namespace param to hypershift dump command to collect agents and infraenv objects

### PR DESCRIPTION
**Pull request by dharaneeshvrd**

### CODE REVIEW

**Comment by dharaneeshvrd** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-power-conformance

**Comment by dharaneeshvrd** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-power-conformance

**Comment by dharaneeshvrd** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-aws-ovn-mce-conformance

File: ci-operator/step-registry/hypershift/mce/dump/hypershift-mce-dump-commands.sh [hunk 1]
```diff
@@ -30,5 +30,6 @@ fi
 CLUSTER_NAME="$(echo -n $PROW_JOB_ID|sha256sum|cut -c-20)"
 /tmp/hs-cli/hypershift dump cluster --artifact-dir=$ARTIFACT_DIR \
 --namespace local-cluster \
+--agent-namespace "local-cluster-${CLUSTER_NAME}" \
```
Comment (LiangquanLi930) [inline ci-operator/step-registry/hypershift/mce/dump/hypershift-mce-dump-commands.sh:33]: for `--agent-namespace`, we need to verify whether the hosted cluster platform is `agent`
```
oc get hostedclusters -n local-cluster ${CLUSTER_NAME} -ojsonpath="{.spec.platform.type}"
```

**Comment by dharaneeshvrd** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-power-conformance

**Comment by dharaneeshvrd** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-aws-ovn-mce-conformance

**Comment by dharaneeshvrd** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-power-conformance

File: ci-operator/step-registry/hypershift/mce/dump/hypershift-mce-dump-commands.sh [hunk 2]
```diff
@@ -28,7 +28,12 @@ else
...
---name="${CLUSTER_NAME}"
+DUMP_PARAMS="--artifact-dir=${ARTIFACT_DIR} --namespace local-cluster --dump-guest-cluster=true --name=${CLUSTER_NAME}"
+
+PLATFORM_TYPE=$(oc get hostedclusters -n local-cluster ${CLUSTER_NAME} -ojsonpath="{.spec.platform.type}")
+if (( $PLATFORM_TYPE == "agent" )); then
+  DUMP_PARAMS+=(" --agent-namespace \"local-cluster-${CLUSTER_NAME}\"")
+fi
+
+/tmp/hs-cli/hypershift-no-cgo dump cluster "${DUMP_PARAMS[@]}"
+
```
Comment (LiangquanLi930) [inline ci-operator/step-registry/hypershift/mce/dump/hypershift-mce-dump-commands.sh:39]: We had better change to 
```
CLUSTER_NAME="$(echo -n $PROW_JOB_ID|sha256sum|cut -c-20)"
EXTRA_ARGS=""
PLATFORM_TYPE=$(oc get hostedclusters -n local-cluster ${CLUSTER_NAME} -ojsonpath="{.spec.platform.type}")
if (( $PLATFORM_TYPE == "Agent" )); then
  EXTRA_ARGS="${EXTRA_ARGS} --agent-namespace local-cluster-${CLUSTER_NAME}"
fi

/tmp/hs-cli/hypershift-no-cgo dump cluster ${EXTRA_ARGS} \
--artifact-dir=$ARTIFACT_DIR \
--namespace local-cluster \
--dump-guest-cluster=true \
--name="${CLUSTER_NAME}"
```

**Comment by dharaneeshvrd** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-power-conformance

**Comment by dharaneeshvrd** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-power-conformance

File: ci-operator/step-registry/hypershift/mce/dump/hypershift-mce-dump-commands.sh (hunk 2 above)
Comment (dharaneeshvrd) [reply ci-operator/step-registry/hypershift/mce/dump/hypershift-mce-dump-commands.sh:42]: Done.

**Comment by dharaneeshvrd** [comment]
@LiangquanLi930 can you please take a look now?

**Comment by LiangquanLi930** [comment]
i find the error log
```
Link to step on registry info site: https://steps.ci.openshift.org/reference/conformance-tests
Link to job on registry info site: https://steps.ci.openshift.org/job?org=openshift&repo=hypershift&branch=release-4.14&test=e2e-aws-ovn-mce-conformance&variant=periodics, "e2e-aws-ovn-mce-conformance" post steps failed: "e2e-aws-ovn-mce-conformance" pod "e2e-aws-ovn-mce-conformance-hypershift-mce-dump" failed: could not watch pod: the pod ci-op-jtvxmml1/e2e-aws-ovn-mce-conformance-hypershift-mce-dump failed after 15s (failed containers: test): ContainerFailed one or more containers exited
Container test exited with code 1, reason Error
---
MCE version is greater than or equal to 2.4, need to extract HyperShift cli
/tmp/.dockerconfigjson
/bin/bash: line 31: CLUSTER_NAME: unbound variable
{"component":"entrypoint","error":"wrapped process failed: exit status 1","file":"k8s.io/test-infra/prow/entrypoint/run.go:84","func":"k8s.io/test-infra/prow/entrypoint.Options.internalRun","level":"error","msg":"Error executing test process","severity":"error","time":"2024-02-20T09:57:10Z"}
error: failed to execute wrapped command: exit status 1
```
form the test job

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-aws-ovn-mce-conformance

**Comment by dharaneeshvrd** [comment]
@LiangquanLi930 
Not sure whether you are referring to an old instance of the rehearse job. In latest one I can see its passed and collected the dump successfully 
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/48938/rehearse-48938-periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-power-conformance/1760566883087552512/artifacts/e2e-mce-power-conformance/hypershift-mce-dump/build-log.txt

**Comment by LiangquanLi930** [comment]
> @LiangquanLi930 Not sure whether you are referring to an old instance of the rehearse job. In latest one I can see its passed and collected the dump successfully https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/48938/rehearse-48938-periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-mce-power-conformance/1760566883087552512/artifacts/e2e-mce-power-conformance/hypershift-mce-dump/build-log.txt

yep, this is power job, so let me rehearse one aws job to test it.

**Pull request merged**